src/Neo/SmartContract/Native/CryptoLib.cs

```diff
@@ -98,5 +104,19 @@ public static bool VerifyWithECDsa(byte[] message, byte[] pubkey, byte[] signatu
                 return false;
             }
         }
+
+        // This is for solving the hardfork issue in https://github.com/neo-project/neo/pull/3209
+        [ContractMethod(true, Hardfork.HF_Cockatrice, CpuFee = 1 << 15)]
+        public static bool VerifyWithECDsa(byte[] message, byte[] pubkey, byte[] signature, NamedCurve curve)
```

Comment:
You may safely get rid of `NamedCurve` type here and use `NamedCurveHash`, just check that `curve` falls within the expected range (22 or 23) and throw if not. You can do this because `NamedCurveHash` is compatible. If you implement this, then no `'NamedCurve' is obsolete` linter warnings will be raised.

Reply:
This is a hotpatch, i prefer to keep it simple and use original logic as much as possible.

Reply:
Every TODO is an issue, otherwise it won't be done. See https://github.com/neo-project/neo/issues/3236.